source/extensions/filters/http/buffer/buffer_filter.cc

```diff
@@ -23,8 +23,8 @@ BufferFilterSettings::BufferFilterSettings(
     const envoy::config::filter::http::buffer::v2::Buffer& proto_config)
     : disabled_(false),
       max_request_bytes_(static_cast<uint64_t>(proto_config.max_request_bytes().value())),
-      max_request_time_(
-          std::chrono::seconds(PROTOBUF_GET_SECONDS_REQUIRED(proto_config, max_request_time))) {}
+      max_request_time_(std::chrono::seconds(
```

Comment:
The timer here needs to become optional for the duration of the deprecation. Basically, either make this an absl::optional, or use a default of 0 to indicate the timer is disabled, then don't arm the timer in the rest of the code. If possible please also add a test for this case to make sure there is no timer.

Reply:
This doesn't make an assertion that the timer value is set; the value should default to 0 when not set. I will look into testing to ascertain that.

Reply:
My point is that you need to actually not enable the timer if it's not set. That is not done currently since the timer must be > 0 in the current code.